sdk/docs/manually-written/sdk/component-howtos/application-development/daml-codegen-java.rst

```diff
@@ -1,37 +1,134 @@
+
+Download
+-------------
+
+The Daml codegen is part of the Daml Assistant. :ref:`Download the Daml Assistant <assistant-manual-managing-releases>`.
+
+.. TODO(#347) Fix the link to link to Download the Daml Assistant of "Daml Assistant"
+
+Install
+-------------
+
+The Daml codegen is part of the Daml Assistant.
+
+.. TODO(#347) Add link "Download the Daml Assistant" to point to download section of "Daml Assistant"
+
+Configure
+-------------
+
+There are two modes to configure the Daml codegen:
+
+- Command line configuration, specifying **all** settings in the command line.
+
+- Project file configuration, specifying **all** settings in the ``daml.yaml``.
+
+Command line configuration
+--------------------------
+
+Help for Daml codegen for Java::
+
+  $ daml codegen java --help
+
+Codegen command line configuration settings::
+
+      <DAR-file[=package-prefix]>...
+                               DAR file to use as input of the codegen with an optional, but recommend, package prefix for the generated sources.
+      -o, --output-directory <value>
+                               Output directory for the generated sources
+      -d, --decoderClass <value>
+                               Fully Qualified Class Name of the optional Decoder utility
+      -V, --verbosity <value>  Verbosity between 0 (only show errors) and 4 (show all messages) -- defaults to 0
+      -r, --root <value>       Regular expression for fully-qualified names of templates to generate -- defaults to .*
+      --help                   This help text
 
+Project file configuration
+--------------------------
 
-Generate Java Code from Daml
-############################
+The above settings can be configured in the ``codegen`` element of the Daml project file ``daml.yaml``.
 
-Introduction
-============
+Here is an example::
 
-When writing applications for the ledger in Java, you want to work with a representation of Daml templates and data types in Java that closely resemble the original Daml code while still being as true to the native types in Java as possible. To achieve this, you can use Daml to Java code generator ("Java codegen") to generate Java types based on a Daml model. You can then use these types in your Java code when reading information from and sending data to the ledger.
+    sdk-version: 2.0.0
+    name: quickstart
+    source: daml
+    init-script: Main:initialize
+    parties:
+      - Alice
+      - Bob
+      - USD_Bank
+      - EUR_Bank
+    version: 0.0.1
+    exposed-modules:
+      - Main
+    dependencies:
+      - daml-prim
+      - daml-stdlib
+    codegen:
+      java:
+        package-prefix: com.daml.quickstart.iou
+        output-directory: java-codegen/src/main/java
+        verbosity: 2
 
-The :ref:`Daml assistant documentation <daml-codegen>` describes how to run and configure the code generator for all supported bindings, including Java.
+Operate
+-------------
```

Comment:
Note: title underlines must match in length.

Reply:
This is not necessary (they should be "[at least as long as the text"](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections))

Reply:
I will cut them though since they are prettier if they match in length.